storage/decorator.py

```diff
@@ -255,43 +270,36 @@ def wrapper(
             json_collector_state.ending_offset = range_start
 
             iterator = func(storage, range_start, body, is_gzipped)
-            # if we know it's a single json we replace body with the whole content,
-            # and mark the object as started
-            if storage.json_content_type == "single" and event_list_from_field_expander is None:
-                single: list[tuple[Union[StorageReader, bytes], int, int, bytes]] = list(
-                    [
-                        (data, starting_offset, ending_offset, newline)
-                        for data, starting_offset, ending_offset, newline, _ in iterator
-                    ]
-                )
-
-                newline = single[0][-1]
-                starting_offset = single[0][1]
-                ending_offset = single[-1][2]
-
-                data_to_yield: bytes = newline.join([x[0] for x in single])
-                yield data_to_yield, starting_offset, ending_offset, newline, None
-            else:
-                for data, starting_offset, ending_offset, newline, _ in iterator:
-                    assert isinstance(data, bytes)
+            # if we know it's a single json we wrap the iterator with _collect_single
+            # and mark the object as json and started
+            if storage.json_content_type == "single":
+                iterator = _collect_single(iterator=iterator)
+                json_collector_state.is_a_json_object = True
+                json_collector_state.has_an_object_start = True
+
+            for data, starting_offset, ending_offset, newline, _ in iterator:
+                assert isinstance(data, bytes)
+
+                # if it's not a json object we can just forward the content by lines
+                if not json_collector_state.has_an_object_start:
```

Comment:
This comment does not seem to be consistent with the if statement below. Should this use `json_collector_state.is_a_json_object` instead?